src/ProtocolZoo/ProtocolZoo.jl

```diff
@@ -212,15 +219,15 @@ end
     end
 end
 
-function findswapablequbits(net, node, pred_low, pred_high) # TODO parameterize the query predicates and the findmin/findmax
+function findswapablequbits(net, node, pred_low, pred_high, choose_low, choose_high) # TODO parameterize the query predicates and the findmin/findmax
     reg = net[node]
 
     low_nodes  = queryall(reg, EntanglementCounterpart, pred_low, ❓; locked=false, assigned=true)
     high_nodes = queryall(reg, EntanglementCounterpart, pred_high, ❓; locked=false, assigned=true)
 
     (isempty(low_nodes) || isempty(high_nodes)) && return nothing
-    _, il = findmin(n->n.tag[2], low_nodes) # TODO make [2] into a nice named property
-    _, ih = findmax(n->n.tag[2], high_nodes)
+    _, il = choose_low(n->n.tag[2], low_nodes) # TODO make [2] into a nice named property
+    _, ih = choose_high(n->n.tag[2], high_nodes)
```

Comment:
oh, you are saying I should have removed the underscore. Indeed, the underscore should not be there.

Reply:
No, in case of a vector of `NamedTuple` the output is the `NamedTuple`, the above snippet takes in a simple vector.

```julia
julia> using QuantumSavory; using QuantumSavory.ProtocolZoo: EntanglementCounterpart;

julia> reg = Register(5);

julia> a = [(slot=reg[9], tag=Tag(EntanglementCounterpart, 4, 1)), (slot=reg[8], tag=Tag(EntanglementCounterpart, 2, 5)), (slot=reg[4], tag=Tag(EntanglementCounterpart, 3, 1)), (slot=reg[1], tag=Tag(EntanglementCounterpart, 3, 1))]
4-element Vector{@NamedTuple{slot::RegRef, tag::Tag}}:
 (slot = Slot 9, tag = Entangled to 4.1)
 (slot = Slot 8, tag = Entangled to 2.5)
 (slot = Slot 4, tag = Entangled to 3.1)
 (slot = Slot 1, tag = Entangled to 3.1)

julia> argmax(n->n.tag[2], a)
(slot = Slot 9, tag = Entangled to 4.1)
```

Reply:
Ah, yes, it should be just `il = choose_low(low_nodes). Then it all works, right? `argmax` has way more signatures than I expected. With that change the random_index function should also work. Apologies for the only partial update.

Reply:
`argmax`(argmin) returns the tuple: (slot = Slot 9, tag = Entangled to 4.1)
we can't use it to index the array of NamedTuple, so it doesn't work like that, if we want to do argmax, we would have to return `il` and `ih` directly

Reply:
`argmax((n->n.tag[2] for n in low_nodes))`

Start from scratch here: what is the point of this change -- it is to make it easy for a user to provide a chooser function that gives you the preferred node to select. You can not have such a function depend on the internals of your code, you can not expect the user to know of it. Apologies for the repeated typos though, those are on me.

Reply:
The change totally makes sense. The problem is just that it the way it was implemented before(using `argmax` or `argmin`), it didn't work with indexing.
The third comment suggests doing `il = choose_low(low_nodes)`, that throws an error:

```julia
ERROR: MethodError: no method matching isless(::RegRef, ::RegRef)

Closest candidates are:
  isless(!Matched::Missing, ::Any)
   @ Base missing.jl:87
  isless(::Any, !Matched::Missing)
   @ Base missing.jl:88
```

Reply:
Check my last comment. Does this work: `argmax((n->n.tag[2] for n in low_nodes))` or something to that effect.

Reply:
The documentation comment also helps with this:

```
the `nodeL` predicate can return many positive candidates; `chooseL` picks one of them (by index into the array of filtered `nodeL` results), defaults to a random pick `arr->rand(keys(arr))`; if you are working on a repeater chain a good choice is `argmin`, i.e. the node furthest to the "left"
```

Reply:
I see, thanks, I got really confused in the beginning, sorry about that